### PR TITLE
tests/kola: add version comparison functions to commonlib.sh

### PR DESCRIPTION
**Pull request by dustymabe**
Leveraging these functions will make it easier to do comparision checks inside of our tests.

Specifically for RPM version comparisons something like /usr/bin/rpmdev-vercmp would be better but that would require having rpmdevtools installed in FCOS, which we don't have.

### CODE REVIEW

**Comment by dustymabe** [comment]
tested this with:

```
$ cat test-sort.sh 
. tests/kola/data/commonlib.sh
compare() {
    A=$1; B=$2
    echo "Comparing $A and $B"
    vereq $A $B && echo "$A is equal to $B"
    verlt $A $B && echo "$A is < $B"
    verlte $A $B && echo "$A is <= $B"
    vergt $A $B && echo "$A is > $B"
    vergte $A $B && echo "$A is => $B"
}

compare "0" "1"
compare "1" "1"
compare "1" "2"
compare "1.2.3.4" "1.2.3.5"
compare "100.1" "2"
compare "253.1" "2002.1"
compare "253.1" "253.1"


$ bash test-sort.sh 
Comparing 0 and 1
0 is < 1
0 is <= 1
Comparing 1 and 1
1 is equal to 1
1 is <= 1
1 is => 1
Comparing 1 and 2
1 is < 2
1 is <= 2
Comparing 1.2.3.4 and 1.2.3.5
1.2.3.4 is < 1.2.3.5
1.2.3.4 is <= 1.2.3.5
Comparing 100.1 and 2
100.1 is > 2
100.1 is => 2
Comparing 253.1 and 2002.1
253.1 is < 2002.1
253.1 is <= 2002.1
Comparing 253.1 and 253.1
253.1 is equal to 253.1
253.1 is <= 253.1
253.1 is => 253.1
```

File: tests/kola/data/commonlib.sh [hunk 1]
```diff
@@ -86,3 +86,26 @@ is_service_active() {
...
 }
+
+# Some functions to support version comparison
+vereq() {
+    [ "$1" == "$2" ]
+}
+
+verlt() {
+    vereq $1 $2 && return 1
+    sorted="$(echo -e "$1\n$2" | sort -V)"
```
Comment (jlebon) [inline tests/kola/data/commonlib.sh]: `local`

File: tests/kola/data/commonlib.sh [hunk 2]
```diff
@@ -86,3 +86,26 @@ is_service_active() {
...
+    [  "$1" == $(echo "$sorted" | head -n 1) ]
```
Comment (jlebon) [inline tests/kola/data/commonlib.sh]: Minor: would read better as

```suggestion
    local lowest="$(echo -e "$1\n$2" | sort -V | head -n 1)"
    [  "$1" == $lowest ]
```

File: tests/kola/data/commonlib.sh [hunk 3]
```diff
@@ -86,3 +86,26 @@ is_service_active() {
     # return actual result
     systemctl is-active "${service}"
 }
+
+# Some functions to support version comparison
+vereq() {
+    [ "$1" == "$2" ]
+}
+
+verlt() {
```
Comment (jlebon) [inline tests/kola/data/commonlib.sh:97]: A comment for this to make it clear that it succeeds if the first arg is less than the second arg could be nice.

**Review by jlebon — DISMISSED** [review]
Overall LGTM

**Comment by dustymabe** [comment]
@jlebon - pushed an update that should address those comments.

**Review by Adam0Brien — DISMISSED** [review]
LGTM

**Comment by dustymabe** [comment]
Looks like my new comments like:

```
# Returns true if $1 is less than or equal to $2
```

tripped up https://github.com/coreos/fedora-coreos-config/pull/2048/commits/41e87994811e7eb69d708279fd1521dc2ecae4ea. I guess that code doesn't consider if the string it's searching for is part of a comment in bash or not.

**Comment by dustymabe** [comment]
because I'm lazy I just replaced `s/if/iff` in the comments.

**Review by jlebon — APPROVED** [review]

**Pull request merged**